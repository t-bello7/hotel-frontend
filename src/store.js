import { setupListeners } from "@reduxjs/toolkit/dist/query";
import { configureStore } from "@reduxjs/toolkit";
import { hotelApi } from "./services/hotel";
import authReducer from './features/auth/authSlice';

const store = configureStore({
  reducer: {
    [hotelApi.reducerPath]: hotelApi.reducer,
    auth: authReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(hotelApi.middleware),
});

setupListeners(store.dispatch);
export default store;
