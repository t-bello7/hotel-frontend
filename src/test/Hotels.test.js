import { render, screen, act } from "@testing-library/react";
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { Provider } from "react-redux";
import Hotels from "../components/hotels";
import store from "../redux/configStore";
import data from '../components/list.json';

describe("Hotels page rendering", () => {
  afterEach(() => {
    act(() => store.dispatch({
      type: '',
      payload: data
    }));
  });

  it("check hotels page rendering", async () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Routes>
            <Route path="/hotels" element={<Hotels />} />
          </Routes>
          <Hotels />
        </Provider>
      </BrowserRouter>
    );
    expect(screen.getByText("Hotel List")).toMatchSnapshot();
  });

  it("hotel list page has search box", async () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Routes>
            <Route path="/hotels" element={<Hotels />} />
          </Routes>
          <Hotels />
        </Provider>
      </BrowserRouter>
    );
    expect(screen.getByPlaceholderText("Search...")).toMatchSnapshot();
  });

  it("hotel list has Highcrest hotel", async () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Routes>
            <Route path="/hotels" element={<Hotels />} />
          </Routes>
          <Hotels />
        </Provider>
      </BrowserRouter>
    );
    expect(screen.getByText("Highcrest")).toMatchSnapshot();
  });
});
