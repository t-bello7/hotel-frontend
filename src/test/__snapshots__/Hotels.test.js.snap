// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotels page rendering check hotels page rendering 1`] = `
<h1>
  Hotel List
</h1>
`;

exports[`Hotels page rendering hotel list has Highcrest hotel 1`] = `
<h2
  class="card_title"
>
  <i
    aria-hidden="true"
    class="fa fa-building green_color"
  />
    
  Highcrest
</h2>
`;

exports[`Hotels page rendering hotel list page has search box 1`] = `
<input
  class="form_feild"
  placeholder="Search..."
  type="text"
/>
`;
